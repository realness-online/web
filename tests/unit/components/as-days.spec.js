import { shallowMount } from '@vue/test-utils'
import as_days from '@/components/as-days'
import get_item from '@/modules/item'
const fs = require('fs')
const statements_html = fs.readFileSync('./tests/unit/html/statements.html', 'utf8')
const statements = get_item(statements_html).statements
describe('@/components/as-days', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallowMount(as_days)
  })
  it('Renders though is provided nothing', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  it('Renders a list of statements sorted into the days they were created', async () => {
    await wrapper.setProps({ statements })
    expect(wrapper.element).toMatchSnapshot()
    expect([...wrapper.vm.days.entries()].length).toBe(3)
  })
})
