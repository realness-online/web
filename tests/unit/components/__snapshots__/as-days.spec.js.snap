// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@/components/as-days Renders a list of statements sorted into the days they were created 1`] = `
<section
  class="as-days"
>
  <article
    class="day"
  >
    <header>
      <h4>
        Monday, September 30, 2019
      </h4>
    </header>
     
  </article>
</section>
`;

exports[`@/components/as-days Renders though is provided nothing 1`] = `
<section
  class="as-days"
/>
`;
